lib/src/coap_config.dart

```diff
@@ -27,16 +25,6 @@ abstract class DefaultCoapConfig {
   /// The version of the CoAP protocol.
   String get version => 'RFC7252';
 
-  /// The CoAP specification derived from the protocol version.
-  CoapISpec get spec {
```

Comment:
I've only had a very quick glance at this PR, but I think we will need multiple encoders/decoders once we support TCP? Since the TCP wire format is different from UDP.

Reply:
Oh, very good point, I didn't have such a close look into [relevant parts](https://datatracker.ietf.org/doc/html/rfc8323#section-3.2) of RFC 8323 yet. Having a specific encoder/decoder for each transport protocol definitely makes sense :)

Reply:
I now tried to take into account that the existing Codec classes are UDP-specific in the latest commits.